src/include/sof/math/exp_fcn.h

```diff
@@ -0,0 +1,51 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+ *
+ */
+#ifndef __SOF_MATH_BASE_EXPONENTIAL_H__
+#define __SOF_MATH_BASE_EXPONENTIAL_H__
+
+#include <rtos/bit.h>
+#include <stdint.h>
+
+#define BIT_MASK_Q62P2 0x4000000000000000
+#define CONVERG_ERROR 28823037607936LL	// error smaller than 1e-4,1/2 ^ -44.7122876200884
+#define BIT_MASK_LOW_Q27P5 0x8000000
+#define MIN_IN_VAL -1342177280
+#define MAX_OUT_VAL 1244979733
+#define QUOTIENT_SCALE BIT(30)
+#define MAX_IN_VAL 1342177280
+#define TERMS_Q23P9 8388608
+#define MIN_OUT_VAL 56601
+#define LSHIFT_BITS 8192
+
+/* inv multiplication lookup table */
+/* LUT = ceil(1/factorial(b_n) * 2 ^ 63) */
+static const int64_t iv[19] = {
```

Comment:
This should be in the C file ?

Reply:
Seppo suggests that C and HiFi versions share a similar header file.
I believe it is beneficial to consider

Reply:
That's fine to share the same header, but any user of this symbol(table) would duplicate it in each object file. Intention of the math library is to provide a set of utility functions for use by processing modules (i.e. we just want a single copy of the table in the code) so this is better in the C file that exposes the public API.

Reply:
I'm a little bewildered and would be glad with your advice. 
My question is - You want an extra file with a lookup table ? Or restore the lookup table in the current exp fcn.c file ?

Reply:
> @singalsu suggests that C and HiFi versions share a similar header file.

I don't understand this sorry, could you please elaborate?

- How does "Similar .h file" imply "Must have `iv[]` in .h file" ?

- What is the "HiFi version"? AFAIK "HiFi" is a product name and C is a language; I don't understand how they can share a .h file.

Reply:
The lookup table and constant are the identical for the HiFi [https://github.com/thesofproject/sof/pull/6801](https://github.com/thesofproject/sof/pull/6801) and exponential C versions.

There is a conversation on preventing code duplication.

Reply:
> The lookup table and constant are the identical for the HiFi https://github.com/thesofproject/sof/pull/6801 and exponential C versions.

Thanks!

> There is a conversation on preventing code duplication.

Move to .c file and drop `static`, done?

Reply:
done